### PR TITLE
ispc: use cmake for building.

### PR DESCRIPTION
**Pull request by athas**
The Makefile approach is gone in newer versions of ispc.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @aristidb

### CODE REVIEW

File: pkgs/development/compilers/ispc/default.nix [hunk 1]
```diff
@@ -42,25 +40,26 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     mkdir -p $out/bin
-    cp ispc $out/bin
+    cp bin/ispc $out/bin
```
Comment (jonringer) [inline pkgs/development/compilers/ispc/default.nix]: this is making it fail for me
```
  cp: cannot stat 'bin/ispc': No such file or directory
cannot build derivation '/nix/store/02npn6n2amdq38j9ngfm1363n26kz2g0-embree-2.17.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/pcgvbqi8bkv7dpz7wj97clmm9b2h73ly-luxcorerender-2.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/fx5cdsq1rbmwjw510lhdhp49mva1dyzs-env.drv': 3 dependencies couldn't be built
[0 built (1 failed)]
error: build of '/nix/store/fx5cdsq1rbmwjw510lhdhp49mva1dyzs-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/70851
3 package failed to build:
embree2 ispc luxcorerender
```

Comment (jonringer) [reply pkgs/development/compilers/ispc/default.nix]: actually, the default cmake phase makes this irrelevant. just remove the installPhase altogether

File: pkgs/development/compilers/ispc/default.nix [hunk 2]
```diff
@@ -42,25 +40,26 @@ stdenv.mkDerivation rec {
...
 
   checkPhase = ''
-    export ISPC_HOME=$PWD
+    export ISPC_HOME=$PWD/bin
     for target in $testedTargets
     do
       echo "Testing target $target"
       echo "================================"
       echo
+      cd ../
```
Comment (jonringer) [inline pkgs/development/compilers/ispc/default.nix]: you may want to restore the PWD after this checkPhase, or else it puts me into an invalid directory for the install phase
```
installing
install flags: SHELL=/nix/store/l6h4ya0wzb4b8mr0y58k2gh2nhfql4sn-bash-4.4-p23/bin/bash install
ldconfig: Can't open cache file /nix/store/6yaj6n8l925xxfbcd65gzqx3dz7idrnn-glibc-2.27/etc/ld.so.cache
: No such file or directory
 ============================== !!! WARNING !!! ===============================
 Location of LLVM files in your PATH is different than path in LLVM_HOME
 variable (or LLVM_HOME is not set). The most likely this means that you are
 using default LLVM installation on your system, which is very bad sign.
 Note, that ISPC uses LLVM optimizer and is highly dependent on it. We recommend
 using *patched* version of LLVM 3.8. Patches are availible in
 llvm_patches folder. You can build LLVM manually, or run our scripts, which
 will do all the work for you. Do the following:
 1. Create a folder, where LLVM will reside and set LLVM_HOME variable to its
   path.
 2. Set ISPC_HOME variable to your ISPC location (probably current folder).
 3. Run alloy.py tool to checkout and build LLVM:
   alloy.py -b --version=3.8
 4. Add $LLVM_HOME/bin-3.8/bin path to your PATH.
 ==============================================================================
make: *** No rule to make target 'install'.  Stop.
```

Comment (athas) [reply pkgs/development/compilers/ispc/default.nix]: Hm, I thought that would run in a subshell.  Guess I'm forgetting my shell script trivia.

**Comment by athas** [comment]
I pushed new changes.  It now builds well on my NixOS machine (left my Mac at work so can't test there).

**Comment by athas** [comment]
Any news here?  The `ispc` in Nixpkgs is now two releases behind upstream, presumably because the newer `ispc`s have removed the Makefiles used by Nixpkgs.

**Review by jonringer — APPROVED** [review]
LGTM

```
[5 built, 80 copied (1632.2 MiB), 288.9 MiB DL]
https://github.com/NixOS/nixpkgs/pull/70851
3 package built:
embree2 ispc luxcorerender
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build embree2 ispc luxcorerender

**Comment by athas** [comment]
Thanks!

**Pull request merged**